pkg/sdk/extract.go

```diff
@@ -47,13 +47,8 @@ type ExtractRequest interface {
 	// is returned if no argument is specified.
 	Arg() string
 	//
-	// SetStrValue sets the extracted value for the requested field.
-	// This must be used for fields of string value type only.
-	SetStrValue(v string)
-	//
-	// SetU64Value sets the extracted value for the requested field.
-	// This must be used for fields of u64 value type only.
-	SetU64Value(v uint64)
+	// SetValue sets the extracted value for the requested field.
```

Comment:
```suggestion
	// SetValue sets the extracted value for the requested field.
	//
	// The underlying type of v must be compatible with the field type
	// associated to this extract request (as the returned by FieldType()), 
	// otherwise SetValue will panic.
```
Or something similar.